traps.py

```diff
@@ -43,9 +31,15 @@ def _PUTS():
 
 def _IN():
     """input a string"""
-    #get the location from mem to write in it
-    ch = map(input())
-    #implementation to map to memory
+    #get string from user
+    ch_arr = list(map(input().split()))
```

Comment:
Wrong usage of `map`, you need to pass it a transformer function, see [the documentation](https://www.w3schools.com/python/ref_func_map.asp).
Also, `map` should be the highest order function in line 35.
Finally, I think `split` is unnecessary here.